test/system/test_tpm2_unseal.sh

```diff
@@ -106,7 +104,7 @@ if [ $? != 0 ];then
     exit 1
 fi
 
-tpm2_create -g $alg_create_obj -G $alg_create_key -u $file_unseal_key_pub -r $file_unseal_key_priv -I- -c $file_primary_key_ctx -L $file_policy -A $obj_attr <<< $secret
+tpm2_create -g $alg_create_obj -G $alg_create_key -u $file_unseal_key_pub -r $file_unseal_key_priv -I- -c $file_primary_key_ctx -L $file_policy -E <<< $secret
```

Comment:
I can't recall the spec offhand, but is ECC a mandatory or optional requirement of tpm2.0. How does this work on a real tpm2.0 device.

How does this affect #335 ?

Reply:
@williamcroberts it's mandatory according to [TCG PC Client Platform TPM Profile (PTP) Specification (Family “2.0” Level 00 Revision 01.03 v22 May 22, 2017)](https://trustedcomputinggroup.org/wp-content/uploads/TCG_PC_Client_Platform_TPM_Profile_PTP_Specification_Family_2.0_Revision_1.3v22.pdf).

If you look at "Table 2 - PC Client TPM Algorithms", both the TPM_ALG_ECDH (for {en,de}cryption) and TPM_ALG_ECDSA (for signing) are listed as mandatory.

This affects real TPM2.0 devices in the way that generating ECC keys is much faster than generating RSA keys. How much it depends on the chip, but on some I found that creating a RSA primary key can take up to 40 seconds!

You can measure this easily on your system with:

```
$ time tpm2_createprimary -A o -g sha256 -G rsa -C primary.context

CreatePrimary Succeed ! Handle: 0x80ffffff


real    0m7.245s
user    0m0.002s
sys     0m0.058

$ time tpm2_createprimary -A o -g sha256 -G ecc -C primary.context

CreatePrimary Succeed ! Handle: 0x80ffffff


real    0m0.469s
user    0m0.003s
sys     0m0.058s
```

So how this affect issue #355, well is much faster to execute this test on a real TPM2.0. Or put it on another way, using RSA just makes the test to take more time for no good reasons IMHO.

Reply:
Perfect Mandatory was what I needed to know.